components/templates/CodingLayout.tsx

```diff
@@ -0,0 +1,86 @@
+'use client'
+
+import { allLessons, Lesson } from 'contentlayer/generated'
+import { useState, useEffect } from 'react'
+import { useMediaQuery } from 'react-responsive'
+import { Tabs } from 'components/ui/Tabs'
+import { Info } from './Info'
+import { Code } from './Code'
+
+/**
+ * @slug {string} for fetching challenge data
+ * @solutionScriptSigHex {string} answer to the challenge problem
+ * @next {string} link to next part of chapter
+ */
+export default function CodingLayout({ 
+        slug, 
+        solutionScriptSigHex, 
+        next 
```

Comment:
A problem with directly navigating to `next` is that we also need to store the users progress in local storage. I created #86 to standardize this functionality. Either this component can manage saving the new progress, or we let each page handle saving & navigation.

Reply:
I think we can make use of the logic implemented in this PR - https://github.com/saving-satoshi/saving-satoshi/pull/86/files#diff-e70514de5c6bd678a40df2f613dd52e95f73d83528ef734a628f72dffe1938a2

Reply:
I have changed the tab 4 spaces to tab 2 spaces as well as added the logic implemented in PR - https://github.com/saving-satoshi/saving-satoshi/pull/86 to both the terminal-based challenge.